sklearn/tests/test_calibration.py

```diff
@@ -341,3 +342,24 @@ def decision_function(self, X):
     calibrated_clf = CalibratedClassifierCV(MockTensorClassifier())
     # we should be able to fit this classifier with no error
     calibrated_clf.fit(X, y)
+
+
+def test_calibration_pipeline():
+    """Test that calibration works in pre-fit pipeline with transformer,
+    where X is not array-like, sparse matrix or dataframe at the start. See
+    issue #8710"""
+    fake_features = [
+        {'state': 'NY', 'age': 'adult'},
+        {'state': 'TX', 'age': 'adult'},
+        {'state': 'VT', 'age': 'child'},
+    ]
+    labels = [1, 0, 1]
+
+    pipeline = Pipeline([
+                ('vectorizer', DictVectorizer()),
+                ('clf', RandomForestClassifier())
+        ])
+    pipeline.fit(fake_features, labels)
+
+    calib_clf = CalibratedClassifierCV(pipeline, cv='prefit')
+    calib_clf.fit(fake_features, labels)
```

Comment:
We should probably be checking some fitted attribute from the calibrated classifier to be sure that everything went fine.

Reply:
For instance, I think that `calib_clf.n_features_in_` will not be defined, isn't it?

We will need to add a property since this is a meta estimator:

```python
    @property
    def n_features_in_(self):
        # For consistency with other estimators we raise a AttributeError so
        # that hasattr() returns False the estimator isn't fitted.
        try:
            check_is_fitted(self)
        except NotFittedError as nfe:
            raise AttributeError(
                "{} object has no n_features_in_ attribute."
                .format(self.__class__.__name__)
            ) from nfe

        return self.calibrated_classifiers_[0].n_features_in_
```

Reply:
What does the `from nfe` do here?

Reply:
@lucyleeow It sets the `__cause__` of the raised exception to point to the original exception. This allows Python 3 to print the original error message and "The above exception was the direct cause..." in tracebacks, for enhanced debugging. There's more info about it in the [documentation of the `raise` statement](https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement) and in [the PEP proposing this feature](https://www.python.org/dev/peps/pep-3134/) 🙂

Reply:
Thanks! Good to know!